src/less/inputs.less

```diff
@@ -131,66 +131,61 @@ textarea {
 }
 
 .input-group {
-  display: inline-table !important; // lesshint importantRule: false
+  display: inline-flex;
   width: auto;
-  margin-top: @margin-large;
-  margin-bottom: @margin-large;
+  margin: @margin-large auto;
   border-collapse: separate;
   border-spacing: 0;
 
-  &.block {
-    display: table;
-    width: 100%;
+  input,
+  button,
+  .button,
+  .input-group-addon {
+    border: @input-border;
+    border-left-width: 0;
+    min-height: @input-height;
   }
 
-  & > * { // lesshint universalSelector: false
-    width: 1px;
+  /* IE10 layout fix */
+  div.button {
+    overflow: inherit;
   }
 
-  & > label,
-  & > input,
-  & > select,
-  & > button,
-  & > .input-group-addon,
-  & > .input,
-  & > .select,
-  & > .label,
-  & > .button {
-    display: table-cell;
-    margin-top: 0;
-    margin-bottom: 0;
-    border: @input-border;
-    border-right: @border-none;
-
-    &:first-child {
-      border-top-right-radius: 0;
-      border-bottom-right-radius: 0;
-    }
+  & > *:first-child { // lesshint universalSelector: false
+    border-left: @input-border;
+    border-top-left-radius: @border-radius-base;
+    border-bottom-left-radius: @border-radius-base;
+  }
 
-    &:last-child {
-      border: @input-border;
-      border-top-left-radius: 0;
-      border-bottom-left-radius: 0;
-    }
+  & > *:last-child { // lesshint universalSelector: false
+    border-top-right-radius: @border-radius-base;
+    border-bottom-right-radius: @border-radius-base;
+  }
 
-    &:not(:last-child):not(:first-child) {
-      border-radius: 0;
-    }
+  & > * { // lesshint universalSelector: false
+    display: inline-block;
```

Comment:
no because this is a selector for the children, not the parent. inline-block is fine, and is overridden by the flex declaration on the parent. Have now fixed all IE alignment issues FTW 👍